### PR TITLE
feat: support for daily builds of apisix-base multi-platform images

### PR DESCRIPTION
**Pull request by tzssangglass**

### CODE REVIEW

**Comment by tzssangglass** [comment]
### Things to do before merging this PR:

Set the docker hub account and access token in the Circle CI environment variables, with the keys `DOCKERHUB_USERNAME` and `DOCKERHUB_TOKEN` respectively, just as set this in the Github Action Secret.

### What's not perfect about this PR

The product of this PR refers to this: https://hub.docker.com/r/turien/apisix-base/tags, it will come with two additional products, `dev-arm64` and `dev-x86_64`.

Of course, there is a way to remove it, just add a step at the end to remove the specified tag, ref: https://devopsheaven.com/docker/dockerhub/2018/04/09/delete-docker-image-tag-dockerhub.html, the scripted approach I have tried has failed and with the container approach I have not tried it for two reasons:
1. security
2. this requires granting delete permissions to the docker hub's access token

**Comment by tzssangglass** [comment]
This PR:https://github.com/apache/apisix-docker/pull/314 is based on the current PR build of apisix-base:dev without a platform, and a different platform version of apisix:dev is built in CI: https://github.com/tzssangglass/apisix-docker/runs/6812476882?check_suite_focus=true, build product: https://hub.docker.com/r/turien/apisix/tags

File: .circleci/config.yml [hunk 1]
```diff
@@ -53,6 +53,64 @@ jobs:
...
+          name: submodule
+          command: |
+            git submodule sync
+            git submodule update --init
+
+      - run:
+          name: Build and Push Docker Image
+          command: |
+            echo "$DOCKERHUB_TOKEN" | docker login -u "$DOCKERHUB_USERNAME" --password-stdin
+            docker buildx build -t api7/apisix-base:dev-x86_64 --push \
```
Comment (soulbird) [inline .circleci/config.yml:94]: ditto

File: .circleci/config.yml [hunk 2]
```diff
@@ -53,6 +53,64 @@ jobs:
...
+          name: submodule
+          command: |
+            git submodule sync
+            git submodule update --init
+
+      - run:
+          name: Build and Push Docker Image
+          command: |
+            echo "$DOCKERHUB_TOKEN" | docker login -u "$DOCKERHUB_USERNAME" --password-stdin
+            docker buildx build -t api7/apisix-base:dev-arm64 --push \
```
[the remark this answers is not in the document]
Comment (tzssangglass) [reply .circleci/config.yml]: fixed

File: .circleci/config.yml (hunk 1 above)
Comment (tzssangglass) [reply .circleci/config.yml]: fixed

**Review by soulbird — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Pull request merged**